<?xml version="1.0"?>
<!-- Run SLAM stack -->
<launch>
  <arg name="mapping_algorithm" default="hector_mapping" doc="[gmapping, heactor_mapping]"/>
  <arg name="wp_global_planner" default="false" doc="[true - wp_global_planner, false - global_planner]"/>
  <arg name="x"                 default="15" />
  <arg name="y"                 default="19" />
  <arg name="yaw"               default="0" />

  <include file="$(find wr8_software)/launch/base/move_base.launch">
    <arg name="base_frame"    value="base_link" />
    <arg name="odom_frame"    value="odom"/>
    <arg name="global_frame"  value="map"/>
    <arg name="odom_topic"    value="odom" />
    <arg name="cmd_topic"     value="cmd_vel" />
    <arg name="laser_topic"   value="scan" />
    <arg name="wp_global_planner" value="$(arg wp_global_planner)" />
  </include>

  <include file="$(find wr8_software)/launch/base/$(arg mapping_algorithm).launch">
    <arg name="base_frame"    value="base_link" />
    <arg name="odom_frame"    value="odom"/>
    <arg name="map_resol"     value="0.05"/>
    <arg name="scan_topic"    value="scan" />
  </include>

  <include file="$(find wr8_software)/launch/base/trajectory.launch">
    <arg name="global_frame"  value="map" />
    <arg name="track_frame"   value="base_link"/>
    <arg name="update_rate"   value="10" />
    <arg name="publish_rate"  value="6.0" />
  </include>

  <node pkg="wr8_software" type="set_initial_pose.py" name="set_initial_pose" output="screen">
    <param name="start_x" value="$(arg x)"/>
    <param name="start_y" value="$(arg y)"/>
    <param name="start_yaw" value="$(arg yaw)"/>
  </node>
</launch>
